kong/resty/websocket/client.lua

```diff
@@ -0,0 +1,440 @@
+-- Copyright (C) Yichun Zhang (agentzh)
```

Comment:
We now have our own [fork of lua-resty-websocket](https://github.com/Kong/lua-resty-websocket) for any of our extra features that haven't made it upstream yet.

I think we should try to get this change into the fork and then ship it with OSS (currently it's only packaged in EE, but it is trivial to add to OSS). It is currently installed by [kong-build-tools](https://github.com/Kong/kong-build-tools/blob/0f9ecf613a287da09e3f4166ebd71529bf04bf5c/openresty-build-tools/kong-ngx-build#L323-L338) _in-place of_ `resty.websocket`, so any changes/additions just need to be backwards-compatible, but this hasn't proved to be too much of a challenge yet. Feel free to @/ping me for help and/or code review if needed.

If it's not feasible to do make this change before code freeze, would you be willing to do so in a follow-up PR? I'd like us to not have to maintain multiple versions of this library.

Reply:
@flrgh exactly, vendoring this is not a good idea. RIght now i'm vendoring it so I can test it in a single PR (and
it's easier to tell what did we change).
We already have [patch](https://github.com/Kong/kong-build-tools/blob/master/openresty-patches/patches/1.21.4.1/lua-resty-websocket-0.09_01-client-mtls.patch) for lua-resty-websocket, so adding another one won't bring more burden. So yes I will make this a proper patch and remove the vendoring
after feature freeze.